src/cliffordsynthesis/CliffordSynthesizer.cpp

```diff
@@ -48,8 +48,11 @@ void CliffordSynthesizer::synthesize(const Configuration& config) {
   // SAT problem repeatedly with increasing timestep limits until a satisfying
   // assignment is found. This uses the general SAT encoding without any
   // objective function regardless of the configuration.
-  const auto [lower, upper] = determineUpperBound(encoderConfig);
+  auto [lower, upper] = determineUpperBound(encoderConfig);
 
+  if (lower == 0) {
+    lower = config.minimalTimeSteps;
+  }
```

Comment:
I am not quite sure I agree with this change.
Shouldn't this be something like
```c++
if (config.minimalTimeSteps > 0) {
  lower = std::max(lower, config.minimalTimeSteps);
}
```
which would match the description of the parameter a lot more.

Reply:
Now that I think about it: shouldn't this actually come before the `determineUpperBound` call?
That might actually safe some SAT solver calls there.

Reply:
I added an option to the `EncoderConfig` so the information is available in `determineUpperBound` without changing the funtion signature.